### PR TITLE
CI: Check that UUIDs are unique

### PR DESCRIPTION
**Pull request by SaschaMann**

### CODE REVIEW

**Comment by SaschaMann** [comment]
@iHiD I think you may have to give the `exercism/maintainers-admin` team access to the repo? The codeowners are not working and the team doesn't show up in the review requests, unlike `exercism/ada` etc.

**Review by petertseng — APPROVED** [review]

File: .github/workflows/ci.yml
Comment (SaschaMann) [inline .github/workflows/ci.yml:49]: I added this check as a step to the schema validation job because, imo, it still validates the correctness of the JSON files, it's just that the schema spec doesn't know the idea of uniqueness.

**Review by ee7 — APPROVED** [review]
Looks correct to me, and I can confirm that `shellcheck` doesn't complain about it.

Maybe it's worth also having a check for "there is no UUID in `problem-specifications` that is in a track `config.json`", just in case somebody "generates" a UUID by copying one from a track repo?

See also https://github.com/exercism/configlet/issues/162, where @iHiD writes:
> uuids should be unique across exercism, so where our tooling can help with that I'd like it to.

For what it's worth, here's a list of the duplicate UUIDs in track `config.json` files, as of right now (ordered by track, then slug)

| Track         | Slug                    | Duplicate UUID                         |
| :------------ | :---------------------- | :------------------------------------- |
| bash          | knapsack                | `ac179b77-98a5-4daf-9773-3d68b6cd8548` |
| crystal       | allergies               | `b306bdaa-438e-46a2-ba54-82cb2c0be882` |
| crystal       | protein-translation     | `00c6f623-2e54-4f90-ae3f-07e493f93c7c` |
| dart          | beer-song               | `50c34698-7767-42b3-962f-21c735e49787` |
| dart          | isbn-verifier           | `c52dfe2b-5634-4c0c-8ea2-3ca86e5d9a33` |
| dart          | resistor-color-duo      | `98ae146a-67f9-49ae-9cd6-00882a374248` |
| elixir        | leap                    | `3d0d82d7-fdff-4dd5-b720-7246317bb023` |
| elixir        | resistor-color          | `12efd520-e3e5-437f-b93e-38af14aa605d` |
| elixir        | two-fer                 | `e7f3280e-0bef-4fac-8a69-cbfa2e5f818a` |
| gleam         | leap                    | `3d0d82d7-fdff-4dd5-b720-7246317bb023` |
| gleam         | resistor-color          | `12efd520-e3e5-437f-b93e-38af14aa605d` |
| gleam         | two-fer                 | `e7f3280e-0bef-4fac-8a69-cbfa2e5f818a` |
| groovy        | isbn-verifier           | `c52dfe2b-5634-4c0c-8ea2-3ca86e5d9a33` |
| groovy        | resistor-color-duo      | `98ae146a-67f9-49ae-9cd6-00882a374248` |
| java          | diffie-hellman          | `9b191b2f-b1a4-4c0e-b911-36666bff614d` |
| java          | kindergarten-garden     | `0b92ffee-c092-4ab1-ad78-76c8cf80e1b5` |
| java          | knapsack                | `ac179b77-98a5-4daf-9773-3d68b6cd8548` |
| java          | leap                    | `7c8294ee-5924-4bf8-a72f-31d0e2d7d9a0` |
| java          | tree-building           | `aeaed0e4-0973-4035-8bc5-07480849048f` |
| javascript    | leap                    | `7c8294ee-5924-4bf8-a72f-31d0e2d7d9a0` |
| kotlin        | diffie-hellman          | `9b191b2f-b1a4-4c0e-b911-36666bff614d` |
| kotlin        | kindergarten-garden     | `0b92ffee-c092-4ab1-ad78-76c8cf80e1b5` |
| python        | tree-building           | `aeaed0e4-0973-4035-8bc5-07480849048f` |
| ruby          | allergies               | `b306bdaa-438e-46a2-ba54-82cb2c0be882` |
| ruby          | beer-song               | `50c34698-7767-42b3-962f-21c735e49787` |
| ruby          | grep                    | `99485900-5d16-4848-af1c-2f1a62ad35ab` |
| ruby          | protein-translation     | `00c6f623-2e54-4f90-ae3f-07e493f93c7c` |
| tcl           | grep                    | `99485900-5d16-4848-af1c-2f1a62ad35ab` |

**Comment by SaschaMann** [comment]
> Maybe it's worth also having a check for "there is no UUID in problem-specifications that is in a track config.json", just in case somebody "generates" a UUID by copying one from a track repo?

I don't see how that would cause any problems, but yeah, switching to a check like that would be ideal once `configlet` supports it.

**Comment by iHiD** [comment]
> @iHiD I think you may have to give the exercism/maintainers-admin team access to the repo? The codeowners are not working and the team doesn't show up as an option in the review requests, unlike exercism/ada etc.

Correct. Thanks. Done.

> I don't see how that would cause any problems,

Any two duplicate UUIDs in the whole of Exercism should be considered something that could break something downsteam.

**Comment by SaschaMann** [comment]
> Any two duplicate UUIDs in the whole of Exercism should be considered something that could break something downsteam.

Do you think this PR is better than nothing or should we wait with merging things before configlet has that functionality?

File: .github/workflows/ci.yml
Comment (ee7) [inline .github/workflows/ci.yml:53]:
```suggestion
              jq -r '.. |."uuid"? | select(. != null)' |          # select all UUIDs
```
Just in case we wanted some `jq` voodoo.

For me, the `sed` is more familiar, but the `jq` version seems nice.

**Review by ee7 — COMMENTED** [review]
`jq`, anyone?

**Comment by iHiD** [comment]
Definitely should be taking small steps if they're available, yeah. So getting all checks in as soon as possible is good. I only raised that as a general point so that people understand that overlapping UUIDs **can** be harmful.

**Review by ErikSchierboom — APPROVED** [review]
I think we should merge this and look at the global UUID stuff later. 

I'll leave it up to you @SaschaMann if you want to use the jq suggestion.

**Pull request merged**
